src/test/java/teammates/test/driver/GmailServiceMaker.java

```diff
@@ -0,0 +1,113 @@
+        }
+    }
+
+    private final String username;
+    private final boolean useFreshCredentials;
+
+    GmailServiceMaker(String username) {
+        this(username, false);
+    }
+
+    GmailServiceMaker(String username, boolean useFreshCredentials) {
+        this.username = username;
+        this.useFreshCredentials = useFreshCredentials;
+    }
+
+    /**
+     * Build and return an authorized Gmail client service.
+     * @return an authorized Gmail client service
+     */
+    Gmail makeGmailService() throws IOException {
+        Credential credential = authorizeAndCreateCredentials();
+        return new Gmail.Builder(HTTP_TRANSPORT, JSON_FACTORY, credential)
+                .setApplicationName("teammates")
+                .build();
+    }
+
+    /**
+     * Authorizes the user and creates an authorized Credential.
+     * @return an authorized Credential
+     */
+    private Credential authorizeAndCreateCredentials() throws IOException {
+        GoogleClientSecrets clientSecrets = loadClientSecretFromJson();
+
+        GoogleAuthorizationCodeFlow flow = buildFlow(clientSecrets);
+
+        if (useFreshCredentials) {
+            flow.getCredentialDataStore().clear();
+        }
+
+        if (flow.getCredentialDataStore().isEmpty()) {
+            System.out.println("Please login as: " + username);
+        }
+
+        return getCredentialFromFlow(flow);
+    }
+
+    private GoogleClientSecrets loadClientSecretFromJson() throws IOException {
+        InputStream in = new FileInputStream(new File(TestProperties.TEST_GMAIL_API_FOLDER, "client_secret.json"));
+        return GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+    }
+
+    private GoogleAuthorizationCodeFlow buildFlow(GoogleClientSecrets clientSecrets) throws IOException {
+        // if the scopes ever need to change, the user will need to manually delete the credentials of the username found in
+        // <TestProperties.TEST_GMAIL_API_FOLDER>/<username>
+        final List<String> scopes = Arrays.asList(GmailScopes.GMAIL_READONLY, GmailScopes.GMAIL_MODIFY);
+        FileDataStoreFactory dataStoreFactory = new FileDataStoreFactory(
+                new File(TestProperties.TEST_GMAIL_API_FOLDER, username));
+        return new GoogleAuthorizationCodeFlow.Builder(
+                HTTP_TRANSPORT, JSON_FACTORY, clientSecrets, scopes)
+                .setDataStoreFactory(dataStoreFactory)
+                .setAccessType("offline")
+                .build();
+    }
+
+    /**
+     * Gets the credential containing the access token from the flow if it exists. Otherwise a local server receiver is used
+     * to receive authorization code and then exchanges the code for an access token.
+     */
+    private Credential getCredentialFromFlow(GoogleAuthorizationCodeFlow flow) throws IOException {
+        return new AuthorizationCodeInstalledApp(
+                flow, new LocalServerReceiver()).authorize("user");
```

Comment:
Why not use `username` as the ID here to avoid the `"user"` magic string? I believe this will also allow usage of just one `StoredCredential` file for all the accounts - the multiple folders per account can then be avoided. When `useFreshCredentials` is true, delete just that user from the credential data store instead of clearing the entire store.